src/Infrastructure/Stize.Infrastructure.Azure/Networking/ApplicationSecurityGroupExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Pulumi;
+using Pulumi.AzureNextGen.Network.Latest;
+
+namespace Stize.Infrastructure.Azure.Networking
+{
+    public static class ApplicationSecurityGroupExtensions
+    {
+        /// <summary>
+        /// Sets the resource group the <see cref="ApplicationSecurityGroup" /> will be created on
+        /// </summary>
+        /// <param name="builder"><see cref="ApplicationSecurityGroup" /> Builder</param>
+        /// <param name="resourceGroup">Resource group name</param>
+        /// <returns></returns>
+        public static ApplicationSecurityGroupBuilder ResourceGroup(this ApplicationSecurityGroupBuilder builder, Input<string> resourceGroup)
+        {
+            builder.Arguments.ResourceGroupName = resourceGroup;
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets the builder name. If the builder has an RandomId associated, 
+        /// appends the hex value of the RandomId to the end of the name
+        /// </summary>
+        /// <param name="builder">Builder instance</param>
+        /// <param name="name">Builder name</param>
+        /// <returns>The builder argument</returns>
+        public static ApplicationSecurityGroupBuilder Name(this ApplicationSecurityGroupBuilder builder, Input<string> name)
+        {
+            if (builder.RandomId != null)
+            {
+                builder.Arguments.ApplicationSecurityGroupName = builder.RandomId.Hex.Apply(r => $"{name}-{r}");
```

Comment:
I recently found that the logic for creating the random names was wrong. Please find a corrected example:

https://github.com/stize/infrastructure/blob/0122c548778a1a6789fd91ca8b3368be2f6af32b/src/Infrastructure/Stize.Infrastructure.Azure/ResourceGroupExtensions.cs#L18

Use the same approach here please.

The reason being, name and Hex are both Pulumi Input values. We need to Apply both (in a concatenation) in order for the evaluated values to be passed to the resource name.